enhancements/network/sdn-live-migration.md

```diff
@@ -0,0 +1,358 @@
+`status.networkType` field of the `network.config` CR. It will trigger MCO to
+apply new MachineConfig to each node:
+1. MCO will 
+   - rerender the MachineConfig for each node
+   - try to cordon and drain the node one node
+2. CNO finds a node is being drained by MCO by watching the MCO node
+   annotations, it will:
+   - pause the node reboot by annotate the node with
+     `machineconfiguration.openshift.io/rebootPausedBy: network-operator`.
+3. MCO will set node annotation `machineconfiguration.openshift.io/state:
+   RebootPending`, when draining node is completed.
+4. CNO finds the node has been drained by MCO, it will:
+   - annotate the node with ovnkube annotations:
+     `k8s.ovn.org/node-subnets: '{"default":"10.131.0.0/23"}'`
+   - remove the "migration.network.openshift.io/plugin" label from the node wait
+     for the sdn-node DaemonSet to decrement its NumberReady
+   - label the node "migration.network.openshift.io/plugin: OVNKubernetes"
+   - remove the `rebootPausedBy` node annotation.
+5. MCO will reboot the node
+6. After booting up, br-ex will be created and ovnkube-node will run in hybrid
+   overlay mode on the node.
+7. MCO will uncordon the node. Pods will be created on the node using
+   ovn-kubernetes as the default CNI plugin.
+
+The above process will be repeated for each node, until all the nodes have been
+applied to the new MachineConfig and converted to OVN Kubernetes.
+
+#### Migration Cleanup
+
+Once migration is complete, CNO will:
+- delete the openshift-sdn DaemonSets
+- redeploy ovn-kubernetes in “normal” mode (no migration mode config, no node
+  affinity).
+- remove the migration-related labels from the nodes
+
+### API
+
+To start the migration, users need to update the `network.operator`
+CR by adding:
+
+```json
+{ 
+  "spec": { 
+    "migration": {
+      "networkType": "OVNKubernetes",
+      "type": "live"
+    }
+  } 
+}
+```
+
+On removing of the `spec.migration` field, CNO will start the migration cleanup.
+
+CNO will also report the migration state to under the `status` of
+`network.operator` CR.
+
+```json
+{ 
+  "status": { 
+    "migration": {
+      // The state can be 'Setup', 'Working', 'Done' or 'Error', 
+      "state": "Working",
+      // the reason needs to be filled when state is 'Error'.
+      "reason": ""
+    }
+  } 
+}
+```
+
+### Lifecycle Management
+
+This is a one-time operation for a cluster, therefore no lifecycle management.
+
+### Test Plan
+
+TBD
+
+### Graduation Criteria
+
+Graduation criteria follows:
+
+#### Dev Preview -> Tech Preview
+
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+
+#### Tech Preview -> GA
+
+- More testing (upgrade, scale)
+- Add CI job
+- Sufficient time for feedback
+
+#### Removing a deprecated feature
+N/A
+
+### Upgrade / Downgrade Strategy
+
+This is a one-time operation for a cluster, therefore no upgrade / downgrade
+strategy.
```

Comment:
A stated goal is "The migration operation shall be able to be rolled back if needed." so how will the rollback be handled?  I don't see a better section to capture this in.